src/components/Item.js

```diff
@@ -0,0 +1,40 @@
+import React, { useState, useEffect } from 'react';
+import '../styles/Item.css';
+
+const Item = ({ item, handleChange }) => {
+  const [checked, setChecked] = useState(false);
+  const className = checked ? 'completed' : '';
+
+  useEffect(() => {
+    const checkDate = () => {
+      // Current date in seconds
+      const now = Date.now() / 1000;
+      // Number of seconds in a day
+      const oneDay = 60 * 60 * 24;
+      // Checks to see if the number of seconds that have elapsed since
+      // the last purchased date is less than one day
+      const isChecked = now - item.lastPurchasedDate.seconds < oneDay;
```

Comment:
I was wondering if it is a good idea to use the moment.js package? It can also help us save the right date format and easily retrieve information for future calculations. (We can all decide this together).

Reply:
Good question. I'm also wondering if we should set the lastPurchasedDate using a serverTimestamp.

https://firebase.google.com/docs/reference/js/firebase.firestore.FieldValue#servertimestamp

@segdeha, @stacietaylorcima or @adriennefriend  do you have any wisdom on working with dates and times w/ firestore?

Reply:
Great question! If you want to work with t[he native Firestore Timestamp](https://firebase.google.com/docs/reference/js/firebase.firestore.Timestamp) type, using Moment or... (trying to remember what that lightweight alternative to Moment is...?) could be helpful!

Another approach would be just to use UNIX or JavaScript timestamps stored as integers. That does imply you doing more math in your code.

Just in case it’s not familiar territory:

- UNIX timestamp is the number of seconds since January 1, 1970
- JavaScript timestamp is the number of milliseconds since January 1, 1970 (what you get when you call `Date.now()`)

Reply:
When you're adding dependencies to the project, you can run them through [Bundlephobia](https://bundlephobia.com/result?p=moment@2.27.0) to see how big they are. `date-fn` is a lightweight alternative as well. Here is how it stacks up to `moment` in size: 

![](https://cdn.zappy.app/c206966519f71bb1c886dcc32f52283b.png)
![](https://cdn.zappy.app/3513d48cd97a54a58c40f0178126c9b8.png
)

Reply:
Ah this is super useful. Thank you!